app/lib/collection-role.coffee

```diff
@@ -0,0 +1,12 @@
+apiClient = require '../api/client'
+
+module.exports =
+  hasSettingsRole: ->
+    if @props.user?
+      apiClient.type('collection_roles').get(collection_id: @props.collection.id, user_id: @props.user.id)
+        .catch ->
+          false
+        .then ([userRoles]) ->
+          userRoles.roles.some (role) -> role.match ///owner|collaborator///i
```

Comment:
won't the .then only run on success? if it's response is a 200 and an empty array then it won't match and will return false, no?